rclpy/src/rclpy/duration.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <pybind11/pybind11.h>
+#include <rcl/time.h>
+
+#include <memory>
+
+#include "duration.hpp"
+
+namespace rclpy
+{
+void
+_rclpy_destroy_duration(PyObject * pycapsule)
+{
+  PyMem_Free(PyCapsule_GetPointer(pycapsule, "rcl_duration_t"));
+}
+
+py::capsule
+create_duration(int64_t nanoseconds)
+{
+  auto duration = static_cast<rcl_duration_t *>(PyMem_Malloc(sizeof(rcl_duration_t)));
```

Comment:
@cottsay FYI there's `py::allocate` too.

Reply:
Oof, this pattern is everywhere. Maybe we should start a backlog ticket to sweep through and unify the pattern. I'm sure we can come up with more patterns we'd like to change as well...